src/main/java/com/microsoft/azure/kusto/kafka/connect/sink/TopicPartitionWriter.java

```diff
@@ -103,8 +102,8 @@ public void open() {
                 fileThreshold,
                 this::handleRollFile,
                 this::getFilePath,
-                flushImmediately ? 0 : flushInterval,
-                this.eventDataCompression == null);
+                shouldCompressData ? 0 : flushInterval,
```

Comment:
hmm, why?

Reply:
we shouldnt compress avro,parquet or orc

Reply:
this is a mistake i made after last change that i introduced - shouldCompressData in FileWriter

Reply:
oh i see - its not good now - sorry - will change